node/src/reactor/participating.rs

```diff
@@ -645,103 +704,53 @@ impl reactor::Reactor for Reactor {
                     info!("genesis chainspec name {}", chainspec.network_config.name);
                     info!("genesis state root hash {}", post_state_hash);
                     trace!(%post_state_hash, ?execution_effect);
-                    initial_pre_state = ExecutionPreState::new(
+                    let initial_pre_state = ExecutionPreState::new(
                         0,
                         post_state_hash,
                         Default::default(),
                         Default::default(),
                     );
-                    finalized_block = FinalizedBlock::new(
+                    let finalized_block = FinalizedBlock::new(
```

Comment:
Can't we use the new `create_immediate_switch_block()` function here?

Reply:
Of course! That's why I extracted the method! :facepalm:

Reply:
Done in https://github.com/casper-network/casper-node/pull/2416/commits/c222e53e6cbd8330e944b08ef4ebfe3e0714ec40.